### PR TITLE
Migrate time zone and dayperiod fixtures to neo datetime

### PR DESCRIPTION
**Pull request by sffc**
#1317

### CODE REVIEW

File: components/datetime/tests/patterns/tests/time_zones.json [hunk 1]
```diff
@@ -652,7 +652,7 @@
           "GenericNonLocationShort"
         ],
         "fallback_formats": ["LocalizedGmt"],
-        "expected": ["GMT"]
+        "expected": ["Тролль"]
```
Comment (sffc) [inline components/datetime/tests/patterns/tests/time_zones.json:655]: The changes in this file _appear_ to be "actual bugs" based on my reading of the spec and the locale data.

File: components/datetime/tests/datetime.rs [hunk 2]
```diff
@@ -645,68 +606,26 @@ fn test_time_zone_patterns() {
...
-                    ),
-                    AnyPayloadProvider::from_payload::<MetazoneGenericNamesLongV1Marker>(
-                        metazone_generic_long_data.payload.clone(), //
-                    ),
-                    AnyPayloadProvider::from_payload::<ExemplarCitiesV1Marker>(
-                        exemplar_cities_data.payload.clone(), //
-                    ),
-                ]);
-
+                let parsed_pattern = DateTimePattern::try_from_pattern_str(pattern_input).unwrap();
```
Comment (sffc) [inline components/datetime/tests/datetime.rs:609]: I feel like I deserve a spot bonus for finally getting rid of this terrible pattern override code that's been haunting us for so long

**Review by Manishearth — DISMISSED** [review]

**Comment by sffc** [comment]
Strange, the CI failure doesn't reproduce locally

**Comment by sffc** [comment]
The PR is passing CI; can I have another approval?

Otherwise I might push more commits migrating more tests

**Pull request merged**